pkg/secrets/plugins/file-system/store.go

```diff
@@ -0,0 +1,120 @@
+package filesystem
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"get.porter.sh/porter/pkg/secrets"
+	"get.porter.sh/porter/pkg/secrets/plugins"
+	"get.porter.sh/porter/pkg/secrets/plugins/host"
+	cnabsecrets "github.com/cnabio/cnab-go/secrets"
+	"github.com/hashicorp/go-hclog"
+	"github.com/pkg/errors"
+)
+
+var _ plugins.SecretsPlugin = &Store{}
+
+const SECRET_FOLDER = "secrets"
+
+type Config struct {
+	PathPrefix string `json:"path_prefix"`
+	secretDir  string
+}
+
+// Store implements an file system secrets store for testing and local
+// development.
+type Store struct {
+	config    Config
+	logger    hclog.Logger
+	hostStore cnabsecrets.Store
+}
+
+// NewStore returns a new instance of the filesystem secret store.
+func NewStore(cfg Config, logger hclog.Logger) *Store {
+	s := &Store{
+		config:    cfg,
+		logger:    logger,
+		hostStore: host.NewPlugin(),
+	}
+
+	return s
+}
+
+// Connect implements the Connect method on the secret plugins' interface.
+func (s *Store) Connect() error {
+	var err error
+	porterHomeDir := os.Getenv("PORTER_HOME")
+	if porterHomeDir == "" {
+		porterHomeDir, err = os.UserHomeDir()
+		if err != nil {
+			return errors.Wrap(err, "could not get user home directory")
+		}
+		porterHomeDir = filepath.Join(porterHomeDir, ".porter")
+	}
+
+	pathPrefix := s.config.PathPrefix
+	if pathPrefix == "" {
+		pathPrefix = SECRET_FOLDER
+
+	}
+
+	secretDir := filepath.Join(porterHomeDir, pathPrefix)
+
+	if err := os.MkdirAll(secretDir, os.ModePerm); err != nil && !errors.Is(err, os.ErrExist) {
```

Comment:
Heads up that you shouldn't use `os.ModePerm` anywhere in Porter. We've learned the hard way that it easily leads to giving the wrong set of permissions to files, which in this case when we are dealing with sensitive data, is an even larger problem.

```
$ ls -la ~/.porter/secrets
total 16
drwxr-xr-x   4 carolynvs  staff  128 Apr 22 10:35 .
drwxr-x---  10 carolynvs  staff  320 Apr 22 10:35 ..
-rw-------   1 carolynvs  staff    5 Apr 22 10:35 01G18YMRP1EXPG0D19SKB9RA7Edbpassword
-rw-------   1 carolynvs  staff    5 Apr 22 10:35 01G18YMRP69SWANWY4C64W6SNRdbpassword
```

Right now we have named permission sets in pkg/version.go to avoid accidentally typing the wrong value in hex but none of them are appropriate in this case. 

All files in the directory (including the directory itself) should be `0600` with only the current user having access, not even the group. Let's make some more constants for this use case and indicate its for sensitive files/directories, e.g. `FileModeSensitiveDirectory`, `FileModeSensitiveWritable`.

Reply:
Can you add a unit test that verifies that the files are being created with the appropriate permissions? In general this plugin needs unit tests anyway but let's make sure that the permissions are included when you add them.

Reply:
That makes sense. Thanks for pointing it out! I have added tests